docs/models/advanced/modeldata.md

```diff
@@ -0,0 +1,61 @@
+
+A [`TileEntityRenderer`][ter] or [`TER`][ter] (previously [`TileEntitySpecialRenderer`][ter] or [`TESR`][ter]) is used to render blocks in a way that cannot be represented with a static baked model (JSON, OBJ, B3D, others).
+
+A [`TER`][ter] has direct access to the `TileEntity` that the renderer is used for whereas an [`IBakedModel`][bakedmodel] only has access to the `BlockState` the model represents, the `Direction` which represents the side which is being rendered (or null), an instance of `Random` for randomised textures and the `IModelData` parameter. The [`IBakedModel::getQuads`][getquads] is unaware of where the block is in the world hence where the `IModelData` parameter steps in to provide extra information from the `TileEntity` using `TileEntity::getModelData`, and from the world using `IBakedModel::getModelData`.
+
+Unlike a [`TER`][ter] which renders every frame, an [`IBakedModel`][bakedmodel] only provides the quads for drawing upon request (which occurs when `World::markBlockRangeForRenderUpdate` is called).
+
+Model Data flow
+------------------------
+
+Model data is first created in the `TileEntity::getModelData` method which by default returns an `EmptyModelData` instance meaning that there is no data provided. This is then passed to the [`IBakedModel::getModelData`][getmodeldata] method as the `tileData` parameter. By default, the [`IBakedModel::getModelData`][getmodeldata] returns the `tileData` parameter. The result of [`IBakedModel::getModelData`][getmodeldata] is then the value of the `modelData` parameter in the [`IBakedModel::getQuads`][getquads] function.
+
+Therefore, in either stage (in the `TileEntity::getModelData` or in the [`IBakedModel::getModelData`][getmodeldata]) we can add or set data.
+
+Constructing Model Data
+-----------------------
+
+Similar to the [blockstate](https://mcforge.readthedocs.io/en/1.14.x/blocks/states/) system, a model data instance consists of multiple `ModelProperty`'s which are created using a builder. In your baked model class, create static final `ModelProperty` instances using the type parameter as the data type you wish to store (e.g. `ModelProperty<String>()` for a string value). The `ModelProperty` constructor also provides an additional predicate parameter which can be used to check the validity of a potential value; however this is set to accept all values by default. Then, to create an `IModelData` instance, we use a `ModelDataMap` builder (`ModelDataMap.Builder`) instance which can setup values that will be present using the `withProperty(<PROPERTY>)` and `withInitial(<PROPERTY>, <VALUE>)` methods which initialise a property with an empty value or with a provided value respectively. Finally, once all properties have been defined, the `build` function should be called to return the `IModelData` instance.
```

Comment:
You are providing a "hard reference" of sorts here instead of referencing from inside the docs.

Reply:
Changed the link in the latest version.